handler.go

```diff
@@ -30,6 +30,31 @@ func ChainHandlers(handlers ...ResponseHandler) ResponseHandler {
 	}
 }
 
+// KeepRespBodyHandlers Combine multiple ResponseHandler and ensure that each processor has access to the original response body
+func KeepRespBodyHandlers(handlers ...ResponseHandler) ResponseHandler {
+	return func(r *http.Response) error {
+		for _, h := range handlers {
+			if h == nil {
+				continue
+			}
+
+			var dup io.ReadCloser
+			r.Body, dup = dupReadCloser(r.Body)
```

Comment:
I'm not sure, but I think if one of the handlers only consumed part of the response, the subsequent handlers would only get a partial buffer.

Instead, what if on a new line between line 35 and line 36, it read the whole body into a buffer and reset the buffer on each loop?